test/functional/Valhalla/src/org/openj9/test/valhalla/NestAttributeTest.java

```diff
@@ -398,4 +398,42 @@ static public void testGetNestMembersAPINestMemberWrongPackage() throws Throwabl
 			} catch (IncompatibleClassChangeError e) { }
 		}
 	}
+	
+	@Test
+	static public void testGetNestMembersAPINestMemberWrongNestHost() throws Throwable {
+		CustomClassLoader classloader = new CustomClassLoader();
+		byte[] aBytes = ClassGenerator.generateClass("test/A", new Attribute[] {new NestMembersAttribute(new String[] {"test/B", "test/C", "test/D"})});
+		Class<?> aClazz = classloader.getClass("test.A", aBytes);
+		byte[] bBytes = ClassGenerator.generateClass("test/B", new Attribute[] {new NestHostAttribute("test/A")});
+		Class<?> bClazz = classloader.getClass("test.B", bBytes);
+		byte[] cBytes = ClassGenerator.generateClass("test/C", new Attribute[] {new NestHostAttribute("test/B")});
+		Class<?> cClazz = classloader.getClass("test.C", cBytes);
+		byte[] dBytes = ClassGenerator.generateClass("test/D", new Attribute[] {new NestHostAttribute("test/A")});
+		Class<?> dClazz = classloader.getClass("test.D", dBytes);
+		
+		try {
+			Class<?> nestMembers[] = aClazz.getNestMembers();
+			Assert.fail("should throw exception on getNestMembers");
+		} catch (IncompatibleClassChangeError e) { }		
+	}
+	
+	@Test
+	static public void testGetNestMembersAPINestMemberWrongNestHost2() throws Throwable {
+		CustomClassLoader classloader = new CustomClassLoader();
+		byte[] aBytes = ClassGenerator.generateClass("test/A", new Attribute[] {new NestMembersAttribute(new String[] {"test/C", "test/D"})});
+		Class<?> aClazz = classloader.getClass("test.A", aBytes);
+		byte[] bBytes = ClassGenerator.generateClass("test/B", new Attribute[] {new NestMembersAttribute(new String[] {"test/C"})});
+		Class<?> bClazz = classloader.getClass("test.B", bBytes);
+		byte[] cBytes = ClassGenerator.generateClass("test/C", new Attribute[] {new NestHostAttribute("test/B")});
+		Class<?> cClazz = classloader.getClass("test.C", cBytes);
+		byte[] dBytes = ClassGenerator.generateClass("test/D", new Attribute[] {new NestHostAttribute("test/A")});
+		Class<?> dClazz = classloader.getClass("test.D", dBytes);
+		
+		try {
+			Class<?> nestMembers[] = aClazz.getNestMembers();
+			Assert.fail("should throw exception on getNestMembers");
+		} catch (IncompatibleClassChangeError e) { }	
+		
+		Class<?> nestMembers[] = bClazz.getNestMembers();
```

Comment:
The other two classes should have their `getNestMember` calls validated as well.  In the successful case, this should check the set of nest members is as expected